packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -94,6 +96,44 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  // compare data between currentSyncData and previousSyncData and update it
+  previousSyncData.entries.filter(entry =>
+    _.findIndex(currentSyncData.deletedEntries, o => o.sys.id === entry.sys.id)
+  )
+
+  previousSyncData.assets.filter(asset =>
+    _.findIndex(currentSyncData.deletedAssets, o => o.sys.id === asset.sys.id)
+  )
+
+  // order is important here, fresh data first
+  currentSyncData.entries = _.uniqBy(
+    currentSyncData.entries.concat(previousSyncData.entries),
+    `sys.id`
+  )
+
+  currentSyncData.assets = _.uniqBy(
+    currentSyncData.assets.concat(previousSyncData.assets),
+    `sys.id`
+  )
+  // TODO run link resolution on the up to date data
+
+  /*
+   * on sync, we only get the synced data because there isn't enough data
+   * we want to trick the SDK to merge the data
+   * on initial sync, we save all the data
+   * on subsequent syncs, we get the updated data; merge with the old data
+   * if there are duplicate entries we take newest
+   */
+  const res = client.parseEntries({
+    items: currentSyncData.entries,
+    includes: {
+      assets: currentSyncData.assets,
```

Comment:
The [doc is not very clear](https://contentful.github.io/contentful.js/contentful/7.13.0/ContentfulClientAPI.html#.parseEntries) about what this `includes` is meant to do. If I should guess its meaning from [EntityCollection](https://contentful.github.io/contentful.js/contentful/7.13.0/Entities.html#.EntryCollection) includes the raw form of the links already present in the items.

Is what we are passing plain JSON?

Reply:
Also I forgot to mention this bit generates a **Maximum call stack size exceeded** error on `node_modules/contentful/dist/contentful.node.js walkMutate()`. Probably because what's passed is not a JSON? Not sure.

With this on the latest version of `rich-text-resolve`  (`2.1.36-rich-text-resolve.14`) doesn't work at all.

Reply:
Includes are all entries that are referenced from entries but itself are not part of the result list.


We will use the SDKs internal API here to resolve the custom request we creating out of the latest sync delta and the cached sync API responses.